test/TestAsset/CircularClasses/Y.php

```diff
@@ -9,7 +9,7 @@
 class Y
 {
     // @codingStandardsIgnoreStart
-    public function __construct(Y $y = null)
+    public function __construct(?Y $y = null)
```

Comment:
This is required to prevent a test failure in 8.4: https://www.php.net/manual/en/migration84.deprecated.php#migration84.deprecated.core.implicitly-nullable-parameter